### PR TITLE
Add hatch pattern support to box and poly annotations

### LINKED ISSUE #9929: Add hatch rendering to Box Annotation
the hatch pattern on varea appears/disappears on zoom. I wanted to create a vertical span with hatch pattern. The only way I could find is creating a varea which is very high compared to the data, so it will always span the entire height. When zooming in and out, the hatch pattern can disappear 

bokeh 2.01 on python 3.6.5 (anaconda), win 10
```
from bokeh.plotting import figure, output_file, show

output_file("hatch.html")
length = 200
small = 0.0001

y = [ 20 +0.0001*i for i in range(length)]
x = list(range(length))
p = figure(plot_width=400, plot_height=400)
p.line(x,y)
yLim = 1000000 #try 100 here, it works
p.varea(x=[length*0.25,length*0.75],y1=[-yLim,-yLim],y2=[yLim,yLim],hatch_color="black",hatch_pattern="v",hatch_alpha=1,fill_alpha=0.5)
#p.vbar(x=length*0.5,top=yLim,bottom = -yLim,width=length*0.5,hatch_color="black",hatch_pattern="v",hatch_alpha=1,fill_alpha=0.5) #this one works as well

p.y_range.start = min(y)
p.y_range.end = max(y)
show(p)

```
on start, the hatch pattern is not visible, now zoom out to see the hatch pattern 
it also works when changing the height of the varea to 100 or using the vbar

### PR DESCRIPTION
**Pull request by mattpap**
Originally this PR was supposed to only finish up hatch pattern implementation, but I also resolved some other issues with box and poly annotations. So, this also adds `Model.syncable` property, which allows to disable server sync for entire models. This in turn allows simplification of box/poly annotations' update logic and allows users to override update policy for selection overlays, allowing to get overlays' coordinates (see [this](https://discourse.bokeh.org/t/get-coordinates-from-image-using-lasso-select-tool/7007) discussion). I also finished implementation of `data` units in `PolyAnnotation` and simplified both implementations.

fixes #9929

### CODE REVIEW

**Comment by mattpap** [comment]
Note also that in future (post 2.3) I will drop internal `screen` property in favour proper absolute screen coordinate managment.

File: bokehjs/src/lib/models/annotations/box_annotation.ts [hunk 1]
```diff
@@ -123,11 +125,13 @@ export namespace BoxAnnotation {
     ew_cursor: p.Property<string | null>
     ns_cursor: p.Property<string | null>
     in_cursor: p.Property<string | null>
+    /** @deprecated */
+    render_mode: p.Property<RenderMode>
```
Comment (bryevdv) [inline bokehjs/src/lib/models/annotations/box_annotation.ts:129]: deprecation sholud go in release notes

File: bokehjs/src/lib/core/util/bbox.ts [hunk 2]
```diff
@@ -191,6 +191,24 @@ export class BBox implements Rect {
...
+    })
+  }
+
+  shrink_by(size: number): BBox {
+    return new BBox({
+      left: this.left + size,
+      right: this.right - size,
+      top: this.top + size,
+      bottom: this.bottom - size,
+    })
```
Comment (bryevdv) [inline bokehjs/src/lib/core/util/bbox.ts:209]: these seem amenable to simple unit tests

File: bokeh/model.py [hunk 3]
```diff
@@ -321,6 +321,13 @@ def id(self) -> str:
 
     """)
 
+    syncable: bool = Bool(default=True, help="""
+    Indicates whether this model should be synchronized when updated in a web browser. This
+    is useful to set when dealing with frequently updated objects, which updated values we
+    don't need, reducing unnecessary network traffic. Note setting this will result in an
+    inconsistent state of objects and any callbacks set on this object will not trigger.
```
Comment (bryevdv) [inline bokeh/model.py:328]:
```suggestion
    Whether this model should be synchronized back to a Bokeh server when updated in a 
    web browser. Setting to False may be useful to reduce network traffic when dealing with 
    frequently updated objects whose updated values we don't need.
    
    .. note:: 
        Setting this property to False will prevent any ``on_change`` callbacks on this object 
        from triggering.
```

Comment (bryevdv) [reply bokeh/model.py:328]: Also we should be explicit: JS callbacks will still tigger, or no?

File: bokehjs/src/lib/models/annotations/box_annotation.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/src/lib/models/annotations/box_annotation.ts:129]: This was deprecated way before, just not indicated in bokehjs.

File: bokeh/model.py (hunk 3 above)
Comment (mattpap) [reply bokeh/model.py:328]: This is only about synchronization, so JS callbacks should trigger. I will make this clearer.

**Comment by bryevdv** [comment]
@mattpap very nice PR I only had a couple of comments

**Pull request merged**